sympy/solvers/solveset.py

```diff
@@ -167,6 +168,11 @@ def _invert(f_x, y, x, domain=S.Complexes):
     if x in y.free_symbols:
         raise ValueError("y should be independent of x ")
 
+    if _is_lambert(f_x - y, x):
+        s = _solve_as_lambert(f_x, y, x, domain)
+        if s.has(LambertW):
+            return x, s
+
     if domain.is_subset(S.Reals):
         x1, s = _invert_real(f_x, FiniteSet(y), x)
     else:
```

Comment:
This looks out of place to me.

Why is `_is_lambert` being checked here instead of inside `_invert_real/_invert_complex`?

Reply:
Sorry, but what should be the approach for this ?!
I thought it should go only in this way. Since inversion of the `lambertW` function to other expressions is so easy but I don't think it will be the same in a reverse way. (i.e. from expression to `lambertW` since it has many forms )

Reply:
I don't think it makes any sense since already domain is being passed.

Reply:
I think you want to suggest to me that I should write the above code in both `_invert_real/_invert_complex`?

Reply:
It is obvious if you look at the code that this function is not supposed to handle specific cases. They are supposed to be handled by `_invert_real` and `_invert_complex`. Adding code in the wrong place like this is what leads to everything being so messy. First you add this and then someone else adds something else and then before long there's a mess of different cases being handled in different places.

You need to look at the code and try and understand the intention of why it is organised the way that it is. Duplicating these 4 lines in two different functions is generally something to avoid but making spaghetti code is much worse than a little bit of duplication.